upgrade react-native-live-markdown to 0.1.76

fabOnReact opened the pull request:
### Details

Bump react-native-live-markdown to 0.1.76 to add fix from PR https://github.com/Expensify/react-native-live-markdown/pull/350.

The PR updates the `NSBaselineOffsetAttributeName` for the iOS TextInput`attributedText` based on the following formula:

```objc
CGFloat baseLineOffset = (maximumLineHeight - maximumFontLineHeight) / 2.0;
```
The text is vertically centered in the lineHeight of the TextInput (which corresponds to the height of the cursor). 
The solution was already added in facebook/react-native:

- PR [facebook/react-native#38359](https://github.com/facebook/react-native/pull/38359) fixed the issue on the Old architecture (Paper) in release [0.73.5](https://github.com/facebook/react-native/blob/02f163e4e83c3676216cd9360166e1cb0a8eda5f/packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputShadowView.mm#L161). 
- commit https://github.com/facebook/react-native/commit/70cc27c901aeb447910e30ac3ceac85990d3c16d fixed the issue in the New architecture (Fabric).

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/17767
$ https://github.com/Expensify/App/issues/14445
$ https://github.com/Expensify/App/issues/15640
PROPOSAL: https://github.com/Expensify/App/issues/17767#issuecomment-1525027774

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

Text in some languages gets cut off at the bottom in composer on ios https://github.com/Expensify/App/issues/17767 

1. Go to a report
2. Paste this text in the composer: "समुद्री उल्लू"
3. Verify that the bottom of the character `मु` is not cut off ([before](https://github.com/Expensify/App/assets/24992535/6307a50d-ffc9-45f6-a3ef-e6f50c97a7d5), [after](https://github.com/Expensify/App/assets/24992535/50308cbc-3002-466b-8b62-29316ed73705))

When moving to another line, the word and emoji are close to each other https://github.com/Expensify/App/issues/14445

1. Launch the app
2. Login with any account
3. In the compose box enter a message using tall letters like `j` on the first line and emoji like :skull: (skull) on the second line
4. Verify that the text on the first line and the emojis on the second are vertically centered and not overlapping ([before](https://github.com/Expensify/App/assets/24992535/a6cef898-cf53-4df1-8bd1-655c5688780b), [after](https://github.com/Expensify/App/assets/24992535/2791a740-eb12-4f24-821c-c1c6dddcf59f))

Cursor in the second and following lines touches the symbols of the previous line https://github.com/Expensify/App/issues/15640

1. Open the app
2. Sign in with any account
3. Go to any chat
5. Type tall characters like j, p, q in the first line
6. Move the cursor to the second line and verify that it does not touch the characters (j, p, q) of the first line ([before](https://github.com/Expensify/App/assets/24992535/405d6897-a1d7-4b06-aa75-f96b06c31008), [after](https://github.com/Expensify/App/assets/24992535/fdbcb6da-5068-4e6c-afe5-c3aec52d5b80)) 

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

Not relevant.

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console



### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.


### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<img src="https://github.com/Expensify/App/assets/24992535/f8222eaa-0509-4a1c-a58f-b233cd201380" width="350" />    

</details>

<details>
<summary>Android: mWeb Chrome</summary>

![Screenshot_1716434309](https://github.com/Expensify/App/assets/24992535/bee1d84c-05b3-4c18-b036-bee1fa46c896)


</details>

<details>
<summary>iOS: Native</summary>

#### Text in some languages gets cut off at the bottom in composer on ios 

Issue https://github.com/Expensify/App/issues/17767
Original comment with screenshots https://github.com/Expensify/react-native-live-markdown/pull/350

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/App/assets/24992535/6307a50d-ffc9-45f6-a3ef-e6f50c97a7d5" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/50308cbc-3002-466b-8b62-29316ed73705" width="350" />       |


#### Expensify Composer (When moving to another line, the word and emoji are close to each other):

Issue https://github.com/Expensify/App/issues/14445
Original comment with screenshots https://github.com/Expensify/react-native-live-markdown/pull/350

The difference of baseline is 1.15 (barely noticeable), maximumLineHeight 20, maximumFontLineHeight 17.69

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/App/assets/24992535/a6cef898-cf53-4df1-8bd1-655c5688780b" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/2791a740-eb12-4f24-821c-c1c6dddcf59f" width="350" />       |


| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/App/assets/24992535/91465191-0f4e-4f2d-a826-e7a735e5c104" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/267c12c9-ebd0-4384-b03a-f0af3d773fc9" width="350" />       |

The result below is achieved after setting [textInputCompose.lineHeight: 40](https://github.com/Expensify/App/blob/ac09bc5612cd4da6d3ec1bede187f80c1563fd7f/src/styles/index.ts#L2080).
- baseLineOffset is 11.15
- maximumLineHeight is 40
- maximumFontLineHeight is 17.69

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/c06452f3-a643-4cf7-afcd-6bd83db72ccb" width="350" />      | <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/75a40d52-d13b-445d-897d-60f2be223aec" width="350" />       |

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/0fa41824-9a58-47f7-aefb-8bcb7e51f31b" width="350" />      | <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/e2c00f4f-b299-4333-ba1d-7cf2cd3b4afc" width="350" />       |

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/65edef0e-5821-487a-ae0b-9966803c4ccd" width="350" />      | <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/b929ae27-d645-4093-92cc-64867c890c40" width="350" />       |

#### Chat - Cursor in the second and following lines touches the symbols of the previous line 

Issue https://github.com/Expensify/App/issues/15640
Original Comment with screenshots https://github.com/Expensify/App/issues/15640#issuecomment-2120007053

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/App/assets/24992535/405d6897-a1d7-4b06-aa75-f96b06c31008" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/fdbcb6da-5068-4e6c-afe5-c3aec52d5b80" width="350" />       |




</details>

<details>
<summary>iOS: mWeb Safari</summary>

![Simulator Screenshot - iPhone 15 Pro - 2024-05-23 at 11 42 09](https://github.com/Expensify/App/assets/24992535/7c4a9e90-165e-4993-8700-868435850fbd)

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

<img width="1648" alt="Screenshot 2024-05-23 at 11 01 37 AM" src="https://github.com/Expensify/App/assets/24992535/3c8c7404-a610-419b-add1-c7c9452f4ff9">

</details>

<details>
<summary>MacOS: Desktop</summary>

<img width="1312" alt="Screenshot 2024-05-23 at 11 46 01 AM" src="https://github.com/Expensify/App/assets/24992535/d36a1d6b-2f09-4574-87b3-f1b64f80b02c">


</details>

fabOnReact committed:
upgrade rn-live-markdown to 0.1.76

fabOnReact commented:
<summary>iOS: Native</summary>

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/App/assets/24992535/6307a50d-ffc9-45f6-a3ef-e6f50c97a7d5" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/50308cbc-3002-466b-8b62-29316ed73705" width="350" />       |

</details>

fabOnReact commented:
| Before    | After |
| ----------- | ----------- |
| <img src="" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/f8222eaa-0509-4a1c-a58f-b233cd201380" width="350" />       |

fabOnReact committed:
Merge branch 'main' into update-rn-live-markdown-0176

eVoloshchak commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

eVoloshchak reviewed (commented):
Works well!

Tests section could be improved, all 3 scenarios lack the final step 
(for the first it would look something like "3. Verify the bottom of the text isn't cut off", for the second "Verify that xext entered and emojis on the next line are not close to each other" and so on)

fabOnReact commented:
### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

Text in some languages gets cut off at the bottom in composer on ios https://github.com/Expensify/App/issues/17767 

1. Go to a report
2. Paste this text in the composer: "समुद्री उल्लू"
3. Verify that the bottom of the character `मु` is not cut off ([before](https://github.com/Expensify/App/assets/24992535/6307a50d-ffc9-45f6-a3ef-e6f50c97a7d5), [after](https://github.com/Expensify/App/assets/24992535/50308cbc-3002-466b-8b62-29316ed73705))

When moving to another line, the word and emoji are close to each other https://github.com/Expensify/App/issues/14445

1. Launch the app
2. Login with any account
3. In the compose box enter a message using tall letters like `j` on the first line and emoji like :skull: (skull) on the second line
4. Verify that the text on the first line and the emojis on the second are vertically centered and not overlapping ([before](https://github.com/Expensify/App/assets/24992535/a6cef898-cf53-4df1-8bd1-655c5688780b), [after](https://github.com/Expensify/App/assets/24992535/2791a740-eb12-4f24-821c-c1c6dddcf59f))

Cursor in the second and following lines touches the symbols of the previous line https://github.com/Expensify/App/issues/15640

1. Open the app
2. Sign in with any account
3. Go to any chat
5. Type tall characters like j, p, q in the first line
6. Move the cursor to the second line and verify that it does not touch the characters (j, p, q) of the first line ([before](https://github.com/Expensify/App/assets/24992535/405d6897-a1d7-4b06-aa75-f96b06c31008), [after](https://github.com/Expensify/App/assets/24992535/fdbcb6da-5068-4e6c-afe5-c3aec52d5b80))

fabOnReact commented:
<details>
<summary>iOS: Native</summary>

#### Text in some languages gets cut off at the bottom in composer on ios 

Issue https://github.com/Expensify/App/issues/17767
Original comment with screenshots https://github.com/Expensify/react-native-live-markdown/pull/350

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/App/assets/24992535/6307a50d-ffc9-45f6-a3ef-e6f50c97a7d5" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/50308cbc-3002-466b-8b62-29316ed73705" width="350" />       |


#### Expensify Composer (When moving to another line, the word and emoji are close to each other):

Issue https://github.com/Expensify/App/issues/14445
Original comment with screenshots https://github.com/Expensify/react-native-live-markdown/pull/350

The difference of baseline is 1.15 (barely noticeable), maximumLineHeight 20, maximumFontLineHeight 17.69

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/App/assets/24992535/a6cef898-cf53-4df1-8bd1-655c5688780b" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/2791a740-eb12-4f24-821c-c1c6dddcf59f" width="350" />       |


| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/App/assets/24992535/91465191-0f4e-4f2d-a826-e7a735e5c104" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/267c12c9-ebd0-4384-b03a-f0af3d773fc9" width="350" />       |

The result below is achieved after setting [textInputCompose.lineHeight: 40](https://github.com/Expensify/App/blob/ac09bc5612cd4da6d3ec1bede187f80c1563fd7f/src/styles/index.ts#L2080).
- baseLineOffset is 11.15
- maximumLineHeight is 40
- maximumFontLineHeight is 17.69

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/c06452f3-a643-4cf7-afcd-6bd83db72ccb" width="350" />      | <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/75a40d52-d13b-445d-897d-60f2be223aec" width="350" />       |

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/0fa41824-9a58-47f7-aefb-8bcb7e51f31b" width="350" />      | <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/e2c00f4f-b299-4333-ba1d-7cf2cd3b4afc" width="350" />       |

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/65edef0e-5821-487a-ae0b-9966803c4ccd" width="350" />      | <img src="https://github.com/Expensify/react-native-live-markdown/assets/24992535/b929ae27-d645-4093-92cc-64867c890c40" width="350" />       |

#### Chat - Cursor in the second and following lines touches the symbols of the previous line 

Issue https://github.com/Expensify/App/issues/15640
Original Comment with screenshots https://github.com/Expensify/App/issues/15640#issuecomment-2120007053

| Before    | After |
| ----------- | ----------- |
| <img src="https://github.com/Expensify/App/assets/24992535/405d6897-a1d7-4b06-aa75-f96b06c31008" width="350" />      | <img src="https://github.com/Expensify/App/assets/24992535/fdbcb6da-5068-4e6c-afe5-c3aec52d5b80" width="350" />       |




</details>

fabOnReact commented:
Thanks @eVoloshchak
- I updated the Test section as requested (more details in this [comment](https://github.com/Expensify/App/pull/42454#issuecomment-2132570500))
- I updated the `iOS: Native` section. Some of the screenshot got corrupted (more details in this [comment](https://github.com/Expensify/App/pull/42454#issuecomment-2132571060))

eVoloshchak reviewed (approved):
LGTM!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9283356335) to staging by https://github.com/marcochavezf in version: 1.4.77-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9309257396) to production by https://github.com/puneetlath in version: 1.4.77-11 🚀

platform | result
---|---
🤖 android 🤖|failure ❌
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9309257396) to production by https://github.com/puneetlath in version: 1.4.77-11 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

mountiny commented:
Also want to highlight this bug that was found in the updated version https://expensify.slack.com/archives/C049HHMV9SM/p1717078251523439

if you cannot access here is a video:
1. in iOS native write a message with emoji in it
2. try to select using selection drag
3. try to select using double-tap
4. it does not worr/ does not work well

https://github.com/Expensify/App/assets/36083550/f2c8af5f-b41a-4735-b6c6-d7a9ec596199

fabOnReact commented:
@mountiny it is not related to the changes in my PR https://github.com/Expensify/react-native-live-markdown/pull/350
https://github.com/Expensify/App/assets/24992535/2b082065-3ff9-4c8b-b5d7-05dc6060c5f4

fabOnReact commented:
The issue is similar to the one fixed with https://github.com/facebook/react-native/pull/38679, but re-produced in a new library (react-native-markdown).
